koku/api/settings/test/tag_mappings/test_view.py

```diff
@@ -0,0 +1,288 @@
+
+        # Check that the response data is filtered correctly (with OCP example)
+        url = reverse("tags-mapping") + "?source_type=ocP"  # also testing case sensitivity
+        response = self.client.get(url, **self.headers)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        for item in response.data["data"]:
+            self.assertEqual(item["source_type"], "OCP")
+
+    def test_get_child(self):
+        """Test the get method for the tag mapping Child view"""
+        url = reverse("tags-mapping-child")
+        response = self.client.get(url, **self.headers)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_child_with_filter(self):
+        """Test the get method for the tag mapping Child view with a filter"""
+        url = reverse("tags-mapping-child") + "?source_type=aWs"
+        response = self.client.get(url, **self.headers)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Check that the response data is filtered correctly
+        for item in response.data["data"]:
+            self.assertEqual(item["source_type"], "AWS")
+
+    def test_get_parent(self):
+        """Test the get method for the tag mapping Parent view"""
+        url = reverse("tags-mapping-parent")
+        response = self.client.get(url, **self.headers)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_parent_with_filter(self):
+        """Test the get method for the tag mapping Parent view with a filter"""
+        url = reverse("tags-mapping-parent") + "?source_type=aWs"
+        response = self.client.get(url, **self.headers)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Check that the response data is filtered correctly
+        for item in response.data["data"]:
+            self.assertEqual(item["source_type"], "AWS")
+
+    def test_put_method_invalid_uuid(self):
+        """Test the put method for the tag mapping view with an invalid uuid"""
+        url = reverse("tags-mapping-child-add")
+        data = {"parent": "29f738e4-38f4-4ed8-a9f4-beed48165220", "children": ["29f738e4-38f4-4ed8-a9f4-beed48165229"]}
+
+        response = self.client.put(url, data, format="json", **self.headers)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_put_method_validate_parent(self):
+        """Test if a parent can be added as a child."""
+        url = reverse("tags-mapping-child-add")
+
+        # Adding sample uuids
+        random_uuid_list = self.retrieve_sample_uuids()
+        url = reverse("tags-mapping-child-add")
+        data = {
+            "parent": random_uuid_list[0],
+            "children": [random_uuid_list[1], random_uuid_list[2], random_uuid_list[3]],
+        }
+        response = self.client.put(url, data, format="json", **self.headers)
+
+        if response.status_code == status.HTTP_200_OK:
```

Comment:
This test should be updated to remove this if statement. The above should be written so that the response is always a 200 which ensures the code below is run.